### ISSUE TITLE
Issues with masking

### ISSUE THREAD

**Issue by AlbertoJaenal** [zju3dv/EfficientLoFTR#74]
Hi, 
thanks for sharing your code and your job!

I have been trying your optimized model and had some issues when using a mask. I think that your code was somehow designed for a "constant" mask, see https://github.com/zju3dv/EfficientLoFTR/blob/main/src/loftr/loftr_module/transformer.py#L132:
`mask_h0, mask_w0, mask_h1, mask_w1 = mask0[0].sum(-2)[0], mask0[0].sum(-1)[0], mask1[0].sum(-2)[0], mask1[0].sum(-1)[0]`
so it picks the number of masked items in the first row/columns, but if the shape changes along the axis, results suffer. 
Does it make sense changing it to
`mask_h0, mask_w0, mask_h1, mask_w1 = mask0[0].sum(-2).max(), mask0[0].sum(-1).max(), mask1[0].sum(-2).max(), mask1[0].sum(-1).max()`
so that it picks the maximum number of masked items along the axis?

Then, when building back the cropped features, there is a problem if the mask is not full in both dimensions in https://github.com/zju3dv/EfficientLoFTR/blob/main/src/loftr/loftr_module/transformer.py#L153-L156 and https://github.com/zju3dv/EfficientLoFTR/blob/main/src/loftr/loftr_module/transformer.py#L162. I propose to change it for:
```
if mask_h0 != mask_H0 and mask_w0 == mask_W0:
    feat0 = torch.cat([feat0, torch.zeros(bs, c, mask_H0-mask_h0, mask_W0, device=feat0.device, dtype=feat0.dtype)], dim=-2)
elif mask_h0 == mask_H0 and mask_w0 != mask_W0:
    feat0 = torch.cat([feat0, torch.zeros(bs, c, mask_H0, mask_W0-mask_w0, device=feat0.device, dtype=feat0.dtype)], dim=-1)
else:
    feat0 = torch.cat([feat0, torch.zeros(bs, c, mask_H0-mask_h0, mask_w0, device=feat0.device, dtype=feat0.dtype)], dim=-2)
    feat0 = torch.cat([feat0, torch.zeros(bs, c, mask_H0, mask_W0 -mask_w0, device=feat0.device, dtype=feat0.dtype)], dim=-1)
```
(and same for the 1)
